static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js

```diff
@@ -108,34 +108,34 @@ define([
-                        if (!outbrainBlockedByAds) {
-                            // if only a high priority ad on page then outbrain is merchandise compliant
-                            checkMediator.waitForCheck('isOutbrainMerchandiseCompliant').then(function (outbrainMerchandiseCompliant) {
-                                if (outbrainMerchandiseCompliant) {
-                                    module.load('merchandising');
-                                } else {
-                                    checkMediator.waitForCheck('isUserInNonCompliantAbTest').then(function (userInNonCompliantAbTest) {
-                                        userInNonCompliantAbTest ? module.load('nonCompliant') : module.load();
-                                    });
-                                }
-                            });
-                        }
-                    });
-                }
-            });
-        }
-
-        return Promise.resolve(true);
+        return checkMediator.waitForCheck('isOutbrainDisabled').then(function (outbrainDisabled) {
+            if (!outbrainDisabled) {
+                // if there is no merch component, load the outbrain widget right away
+                return loadInstantly().then(function(shouldLoadInstantly) {
+                    if (shouldLoadInstantly) {
+                        return checkMediator.waitForCheck('isUserInNonCompliantAbTest').then(function (userInNonCompliantAbTest) {
+                            userInNonCompliantAbTest ? module.load('nonCompliant') : module.load();
+                        });
+                    } else {
+                        // if a high priority merch component and low priority merch component on page block outbrain
+                        return checkMediator.waitForCheck('isOutbrainBlockedByAds').then(function(outbrainBlockedByAds) {
+                            if (!outbrainBlockedByAds) {
+                                // if only a high priority merch component on page then outbrain is merchandise compliant
+                                return checkMediator.waitForCheck('isOutbrainMerchandiseCompliant').then(function (outbrainMerchandiseCompliant) {
+                                    if (outbrainMerchandiseCompliant) {
+                                        module.load('merchandising');
+                                    } else {
+                                        return checkMediator.waitForCheck('isUserInNonCompliantAbTest').then(function (userInNonCompliantAbTest) {
+                                            userInNonCompliantAbTest ? module.load('nonCompliant') : module.load();
+                                        });
+                                    }
+                                });
+                            }
+                        });
+                    }
+                });
+            }
```

Comment:
What happens if `outbrainDisabled` or `outbrainBlockedByAds` is truthy? No promise is returned, which might lead to exceptions upstream. This (and a fair amount of nesting) can be avoided here by inverting the conditions and instantly returning a resolved promise:
```
if (outbrainDisabled) {
  return Promise.resolve();
}

return loadInstantly() ...